FrontEnd/src/components/PageMenu/PageMenu.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import MenuItem from '@mui/material/MenuItem';
+import Menu from '@mui/material/Menu';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+const OPTIONS = ['Validate', 'Report', 'Browse'];
+const LOCATIONS = ['/validate', '/report', '/browse'];
+
+export interface PageMenuProps {}
+
+/**
+ * Allows selecting pages from a menu.
+ *
+ * Shows the selected menu item depending on the route pathname.
+ *
+ * Is responsive. Shows a menu on mobile, and shows links on desktop.
+ */
+export default function PageMenu() {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+
+  // The menu can start depending on the pathname.
+  const [selectedIndex, setSelectedIndex] = React.useState(
+    LOCATIONS.indexOf(location.pathname) === -1
+      ? 0
+      : LOCATIONS.indexOf(location.pathname)
+  );
+
+  // If the route changes, then we change the selected menu too.
+  React.useEffect(() => {
+    if (LOCATIONS.indexOf(location.pathname) !== -1) {
+      setSelectedIndex(LOCATIONS.indexOf(location.pathname));
+    }
+  }, [location.pathname]);
+
+  const open = Boolean(anchorEl);
+  const handleClickListItem = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleMenuItemClick = (
+    event: React.MouseEvent<HTMLElement>,
+    index: number
+  ) => {
+    setSelectedIndex(index);
+    setAnchorEl(null);
+    navigate(LOCATIONS[index]);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <>
+      <Box sx={{ flexGrow: 1, display: { xs: 'none', md: 'flex' } }}>
+        {OPTIONS.map((page, index) => (
```

Comment:
This change is great because it allows people to view the menu regardless of screen size, including from a mobile device, but is it possible for the menu to be a dropdown still rather than a spread of buttons? I think the intention was for "Validate," "Report," and "Browse" to be thought of more as modes of operation rather than page destinations, if that makes sense. Though, we can definitely rethink this as need be! Here's a [link](https://github.com/ForestGeoHack/ForestGEO/wiki/ForestGEO-App-Specification#mvr-1) to the mocks so you can see better what I mean.

Reply:
I wonder if they can be navigated to separately, or if they should only be shown one at a time? 

If they're only one at a time... should we take away the ability to select different options from the menu?

If they are supposed to be all selectable, then I think having a set of links is:
- a bit more common (see the mui nav bar documentation as one example where the row of links)
- set of links is a bit usable, because people can see all the options choose one without having to interact multiple times to choose one

Reply:
We decided to stick with this for now, and think it through a bit more later.